Sources/Dip.swift

```diff
@@ -578,3 +579,70 @@ extension DependencyContainer.Tag: Equatable {
   }
 
 }
+
+
+class DefinitionsContainer {
+  // Backing data structure. Its a map of
+  // [ Type.hashValue: [DefinitionKey: _Definition]]
+  // so we have fash lookup of by Type for the definitions that handle it.
+  private var definitionsByType = [Int: [DefinitionKey: _Definition]]()
+  
+  typealias DefinitionKeyValuePair = (DefinitionKey, _Definition)
+  
+  var all: [DefinitionKeyValuePair] {
+    return definitionsByType.reduce([DefinitionKeyValuePair]()) { (acc, arg1) in
```

Comment:
reduce is quite expensive, it creates a copy of the result on each iteration

Reply:
Fixed